server/src/main/java/org/opensearch/gateway/ReplicaShardAllocator.java

```diff
@@ -70,93 +71,112 @@
  * @opensearch.internal
  */
 public abstract class ReplicaShardAllocator extends BaseGatewayShardAllocator {
+    protected boolean shouldSkipFetchForRecovery(ShardRouting shard) {
+        if (shard.primary()) {
+            return true;
+        }
+        if (shard.initializing() == false) {
+            return true;
+        }
+        if (shard.relocatingNodeId() != null) {
+            return true;
+        }
+        if (shard.unassignedInfo() != null && shard.unassignedInfo().getReason() == UnassignedInfo.Reason.INDEX_CREATED) {
+            // if we are allocating a replica because of index creation, no need to go and find a copy, there isn't one...
+            return true;
+        }
+        return false;
+    }
+
+    protected Runnable getShardCancellationAction(
```

Comment:
why this method have to be refactored and mark protected?

The current method name doesn't tell the intend, may be rename to cancelExistingRecoveryForBetterMatch

Reply:
This function is also getting invoked from the `processExistingRecoveries` in the child class we are adding in #8992 
Updated the name in latest commit